jujugui/static/gui/src/app/store/env/base.js

```diff
@@ -27,9 +27,50 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 YUI.add('juju-env-base', function(Y) {
 
-  var module = Y.namespace('juju.environments');
-  var _sessionStorageData = {};
-  var API_USER_TAG = 'user-';
+  const module = Y.namespace('juju.environments');
+  const _sessionStorageData = {};
+  // Define a Juju tags management object.
+  module.tags = {
+    /**
+      Build a tag with the given type and entity name.
+
+      @method build
+      @param {String} type The tag type (for instance tags.USER or tags.CLOUD).
+      @param {String} name The name of the Juju entity.
+      @return {String} The resulting Juju tag.
+    */
+    build: (type, name) => {
+      return `${type}-${name}`;
+    },
+    /**
+      Parse a Juju entity name from the given tag.
+
+      @method parse
+      @param {String} tag The tag associated with the Juju entity.
+      @param {String} prefix The Juju prefix (see module.prefixes).
+      @return {String} The resulting name of the entity (without the prefix).
+    */
+    parse: (type, tag) => {
+      const exp = new RegExp(`^${type}-`);
```

Comment:
This was more of a pondering question - after doing some micro benchmarks, turns out this is quite slow but because this isn't in a hot path I'm ok with letting it slide :)